nextest-runner/src/runner.rs

```diff
@@ -2078,7 +2078,9 @@ where
                 self.run_stats.on_test_finished(&run_statuses);
 
                 // should this run be cancelled because of a failure?
-                let fail_cancel = self.fail_fast && !run_statuses.last_status().result.is_success();
+                let fail_cancel = self
+                    .max_fail
+                    .map_or(false, |mf| self.run_stats.failed >= mf);
```

Comment:
hmm, so `failed` is one of several kinds of test errors -- there's also timed-out and success. I would add a `non_success_count` method to `RunStats` that sums them up and use that here.

Reply:
Yeah, sorry, I meant timed-out and exec errors. Thanks for catching that!